test/lager_test_backend.erl

```diff
@@ -1113,9 +1113,11 @@ error_logger_redirect_setup() ->
     application:load(lager),
     application:set_env(lager, error_logger_redirect, true),
     application:set_env(lager, handlers, [{?MODULE, info}]),
+    application:set_env(lager, suppress_supervisor_start_stop, false),
+    application:set_env(lager, suppress_application_start_stop, false),
     lager:start(),
     lager:log(error, self(), "flush flush"),
-    timer:sleep(100),
+    timer:sleep(1000),
```

Comment:
It was a bad idea to introduce sleeps in these functions. There has to be a cleaner way to signal that we're ready to proceed with tests.  I think the way to do that is to have the `flush` handler emit a message that the flush is completed. But I guess that could be a new enhancement tackled down the road.

I'm thinking like 

```erlang
   Ref = make_ref(),
   gen_event:call(lager_event, ?MODULE, {flush, Ref, self()}),
   receive
       {Ref, flush_completed} -> lager_event
   after ?TEST_TIMEOUT ->
        erlang:error(test_timeout)
   end.
```

I don't know. Maybe that's a terrible idea but there's got to be something better than a sleep.